tempesta_fw/t/functional/helpers/control.py

```diff
@@ -89,7 +91,9 @@ def prepare(self):
         return True
 
     def results(self):
-        return self.requests, self.errors
+        if (self.rate == 0):
+            self.rate = self.requests / self.duration
```

Comment:
As I understood, `rate == 0` used as an uninitialized value. An if we see `rate==0`, we initialize it with actual value. But `rate == 0` is also possible in (vary rare) case of `self.requests == 0`. I think, it's better to use `rate==-1` for uninitialized value